Notesnook.Inbox.API/src/index.ts

```diff
@@ -0,0 +1,119 @@
+import express from "express";
+import _sodium from "libsodium-wrappers-sumo";
+import { z } from "zod";
+
+const NOTESNOOK_API_SERVER_URL = process.env.NOTESNOOK_API_SERVER_URL;
+if (!NOTESNOOK_API_SERVER_URL) {
+  throw new Error("NOTESNOOK_API_SERVER_URL is not defined");
+}
+
+let sodium: typeof _sodium;
+
+const RawInboxItemSchema = z.object({
+  title: z.string().min(1, "Title is required"),
+  pinned: z.boolean().optional(),
+  favorite: z.boolean().optional(),
+  readonly: z.boolean().optional(),
+  archived: z.boolean().optional(),
+  notebookIds: z.array(z.string()).optional(),
+  tagIds: z.array(z.string()).optional(),
+});
+
+interface EncryptedInboxItem {
+  iv: string;
+  alg: string;
+  cipher: string;
+  length: number;
+}
+
+function encryptData(data: string, publicKey: string): EncryptedInboxItem {
+  try {
+    const recipientPublicKey = sodium.from_base64(publicKey);
+    const dataBytes = sodium.from_string(data);
+    const ciphertext = sodium.crypto_box_seal(dataBytes, recipientPublicKey);
+    return {
+      iv: "iv",
+      alg: "X25519 and XSalsa20-Poly1305",
```

Comment:
unsure about what values to provide for `iv` and `alg`

I'm using this as reference for `alg`
> Sealed boxes leverage the crypto_box construction, which uses X25519 and XSalsa20-Poly1305.

https://libsodium.gitbook.io/doc/public-key_cryptography/sealed_boxes#algorithm-details

Reply:
We use this template for `alg` `encryptionAlgorithm-kdfAlgorithm-base64variant`. So according to that is should be: `xsal-x25519-<base64variant>`. The `base64variant` should be the same as provided to libsodium.

Can you also ensure that we can decrypt the encrypted payload using the private-key? Just to be sure.

Reply:
The `iv` will be the `nonce` or null if there's no `nonce`.

Reply:
> The base64variant should be the same as provided to libsodium.

I'm not provided any base64 variant to libsodium. I'm not sure what to add here

Reply:
> Can you also ensure that we can decrypt the encrypted payload using the private-key? Just to be sure.

Yep this has been tested. I'm able to decrypt with my inbox private key via the `crypto_box_seal_open` method

Reply:
We use `base64_variants.URLSAFE_NO_PADDING` almost everywhere so just use that in all cases.